docs: Bug, feature, and issue templates (#1786)
sopel-irc/sopel: .github/pull_request_template.md

### Description
What does this do? What problem/s does it solve? What issue numbers does it
resolve, if any?

### Checklist
- [ ] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [ ] I can and do license this contribution under the EFLv2
- [ ] There are no test or style issues shown by `make qa` (runs both `make

>   quality` and `make test`)

Author: Welllll... The templates just get dumped into the "New PR" description textbox, so they'll be seen by everyone who opens a new PR (or the other templates for new bug/feature)

>   quality` and `make test`)

Author: `- [ ] There are no issues reported by `make qa` (runs `make quality` and `make test`)` is under 90 chars so that'll probably look better either way

### Description
What does this do? What problem/s does it solve? What issue numbers does it

> resolve?

Reviewer: ```suggestion
resolve (if any)?
```

### Checklist
- [ ] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [ ] My patch is based on the correct branch (eg. `6.6.x` or `master`)

> - [ ] I am opening this PR against that branch

Reviewer: We haven't accepted PRs against `6.6.x` in several months. These two items are probably wholly unnecessary; we can handle patches that need to be made against a maintenance branch on a case-by-case basis.

Author: I was trying to come up with a concise way to say "If this is a bugfix, find the version's branch", but eh if you don't mind the additional handholding and people having to go rebase stuff.

> - [ ] I can and do license this code under EFLv2

Reviewer: Nitpick: Contributions aren't necessarily code.

```suggestion
- [ ] I can and do license my contributions under the EFLv2
```

Author: Fair. I said "this" instead of "my" in case a contribution includes things not written by the PR-opener.

> - [ ] `checkstyle.sh` and `pytest_run.py` show no issues

Reviewer: Reference `make qa` (and the sub-steps, `make quality` & `make test`) instead of the scripts themselves.

Author: Is it important to mention `quality`/`test` on their own, or just `qa`?

Reviewer: It's just faster to check `quality` separately. `make test` takes quite a while, even on SSD machines.

Author: Only 30 seconds on my machine, and I think we'll get better compliance with a simpler instruction since we definitely want both run before a PR. If they want to save time while developing, the others are documented somewhere I assume?

Reviewer: Not really, no. Only as targets in the Makefile if one reads it.

And `make quality` takes about 5 sec, so.

Author: Well, that should be fixed (and if we're using `make`, maybe the scripts put under `contrib/`?, but in the meantime I'll mention all three:

### Description
What does this do? What problem/s does it solve? What issue numbers does it
resolve, if any?

### Checklist
- [ ] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [ ] I can and do license this contribution under the EFLv2
- [ ] There are no test or style issues shown by `make qa` (runs both `make
  quality` and `make test`)
- [ ] I have tested the functionality of the things this change touches
